pkg/auth/jwks.go

```diff
@@ -0,0 +1,102 @@
+package auth
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"os"
+	"time"
+
+	"github.com/MicahParks/keyfunc/v3"
+	"github.com/golang-jwt/jwt/v5"
+	"github.com/spf13/pflag"
+)
+
+type JWKS struct {
+	// Endpoint to load the JWKS from.
+	//
+	// Supports schemes http, https or file.
+	Endpoint string `json:"endpoint" yaml:"endpoint"`
+
+	// How long to cache the JWKS for before reloading.
+	CacheTTL time.Duration `json:"cache_ttl" yaml:"cache_ttl"`
+
+	// Timeout for loading the JWKS.
+	Timeout time.Duration `json:"timeout" yaml:"timeout"`
+}
+
+// LoadedJWKS provides a ready to use jwt.KeyFunc for token verification.
+type LoadedJWKS struct {
+	// KeyFunc is the key function to use for verifying JWTs.
+	KeyFunc jwt.Keyfunc
+}
+
+// Load will ensure that the correct KeyFunc is loaded
+// and available as part of the returned LoadedJWKS pointer.
+func (j *JWKS) Load(ctx context.Context) (*LoadedJWKS, error) {
+	endpoint, err := url.Parse(j.Endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse JWKS endpoint: %w", err)
+	}
+
+	switch endpoint.Scheme {
+	case "http", "https":
+		return j.loadRemote(ctx)
+	default:
```

Comment:
Instead of defaulting to a local path, what about adding another branch for `file`, then reject any other schemes as `unsupported scheme`? Otherwise something like `ftp://foo/jwks.json` would be loaded as a local `jwks.json`, but would prefer `unsupported scheme: ftp`?

Reply:
That's a good idea, I can do that